master/buildbot/reporters/github.py

```diff
@@ -109,6 +111,15 @@ def _create_default_generators(self):
             ),
         ]
 
+    @defer.inlineCallbacks
+    def _get_auth_header(self, props: Union[Properties, None]) -> dict[str, str]:
+        if props:
+            token = yield props.render(self.token)
+        else:
+            token = yield self.renderSecrets(self.token)
```

Comment:
I think it's enough to just do `self.renderSecrets(self.token)`. A secret is just a special renderable that gathers data from secret providers. So a renderable that refreshes GITHUB_TOKEN is should work fine here.

Am I missing anything?

Reply:
The normal render function doesn't have access to the project we are about to build. The idea is that we can use the GITHUB_TOKEN for the project rather than a global GITHUB_TOKEN. I am open for other suggestions.